cranelift/codegen/src/machinst/mod.rs

```diff
@@ -215,7 +215,9 @@ pub enum MachTerminator<'a> {
 /// A trait describing the ability to encode a MachInst into binary machine code.
 pub trait MachInstEmit<O: MachSectionOutput> {
     /// Emit the instruction.
-    fn emit(&self, code: &mut O, flags: &Flags);
+    fn emit(&self, code: &mut O, flags: &Flags, state: &mut Self::State);
+    /// Persistent state carried across `emit` invocations.
+    type State: Default + Clone + Debug;
```

Comment:
nit: can you put this before the `emit` function which uses it, to make reading of this code more linear?